### ISSUE TITLE
Compilation error 'touch_pad_filter_start' was not declared in this scope

### ISSUE THREAD

**Issue by arnaudmorin** [AlessandroAU/Chorus32-ESP32LapTimer#127]
Hello,

I cloned the code
Entered the `ESP32LapTimer` folder
Then trying to upload:

```bash
pio run -e BOARD_OLD -t upload

src/Buttons.cpp: In function 'void newButtonSetup()':
src/Buttons.cpp:63:3: error: 'touch_pad_filter_start' was not declared in this scope
   touch_pad_filter_start(BUTTON1);
   ^~~~~~~~~~~~~~~~~~~~~~
src/Buttons.cpp:63:3: note: suggested alternative: 'touch_trigger_src_t'
Compiling .pio/build/BOARD_OLD/src/HardwareConfig.cpp.o
   touch_pad_filter_start(BUTTON1);
   ^~~~~~~~~~~~~~~~~~~~~~
   touch_trigger_src_t
src/Buttons.cpp:64:3: error: 'touch_pad_set_filter_period' was not declared in this scope
   touch_pad_set_filter_period(BUTTON1);
   ^~~~~~~~~~~~~~~~~~~~~~~~~~~
Compiling .pio/build/BOARD_OLD/src/Laptime.cpp.o
*** [.pio/build/BOARD_OLD/src/Buttons.cpp.o] Error 1

```

**Comment by arnaudmorin** [comment]
After removing the faulty buttons.cpp code:

```bash
$ git diff

diff --git a/ESP32LapTimer/src/Buttons.cpp b/ESP32LapTimer/src/Buttons.cpp
index 22621a7..db3205d 100644
--- a/ESP32LapTimer/src/Buttons.cpp
+++ b/ESP32LapTimer/src/Buttons.cpp
@@ -59,11 +59,6 @@ void newButtonSetup() {
 #ifdef USE_NORMAL_BUTTONS
   pinMode(BUTTON1, INPUT_PULLUP);
   pinMode(BUTTON2, INPUT_PULLUP);
-#else
-  touch_pad_filter_start(BUTTON1);
-  touch_pad_set_filter_period(BUTTON1);
-  touch_pad_filter_start(BUTTON2);
-  touch_pad_set_filter_period(BUTTON2);
 #endif
 }


```

it compiles, but finally error because of too big firmware:

```bash
Error: The program size (1448665 bytes) is greater than maximum allowed (1310720 bytes)
```
What should I do?

**Comment by Smeat** [comment]
Sorry I totally missed your Issue. Not sure if you still have this issue, but as seen in #129 others might have this issue as well.

I just tested it and I am unable to reproduce your error.
Steps I ran:
```
git clone https://github.com/AlessandroAU/Chorus32-ESP32LapTimer
cd Chorus32-ESP32LapTimer/ESP32LapTimer
python -m venv venv
./venv/bin/pip install platformio
./venv/bin/pio run
```

PIO output:
```
[..]
RAM:   [==        ]  16.9% (used 55376 bytes from 327680 bytes)
Flash: [========  ]  79.9% (used 1046702 bytes from 1310720 bytes)
Building .pio/build/BOARD_WROOM/firmware.bin
esptool.py v3.1
Merged 1 ELF section
================================================================================= [SUCCESS] Took 9.81 seconds =================================================================================

Environment      Status    Duration
---------------  --------  ------------
BOARD_DEFAULT    SUCCESS   00:00:10.805
BOARD_TTGO_LORA  SUCCESS   00:00:10.262
BOARD_OLD        SUCCESS   00:00:01.141
BOARD_WROOM      SUCCESS   00:00:09.808
================================================================================= 4 succeeded in 00:00:32.015 =================================================================================
./venv/bin/pio run  96,06s user 15,07s system 345% cpu 32,124 total
```
All Targets are using less than 80% of the available size. I only encountered size issues after enabling Bluetooth, so I am not sure what the cause might be.

Do you have any additional information about your setup? What pio versions are you using?
I used pio version `6.1.3` and `6.0.2` to test this.

**Comment by Z1gun** [comment]
I have pio 6.1.3 and got same issues on Windows and Linux machine. What the correct solution of Buttons.cpp compiler error?
Sorry for off topic, how you get pio in venv?

**Comment by Smeat** [comment]
I was using pip. Sorry, forgot to add that. Fixed it in my previous post.

> What the correct solution of Buttons.cpp compiler error?

That is the weird part: I don't get any error on my machine.

But I was able to reproduce both in a VM. It is probably some library update. I'll look into it in the following ~~days~~ minutes.

@Z1gun can you try and edit line 12 in `platformio.ini` to
```ini
platform = espressif32 @ ~3.5.0
```
This fixes the problem for me.

**Comment by Smeat** [comment]
This should be fixed with #130. If not feel free to reopen.
